pkg/orchestration/controller.go

```diff
@@ -85,65 +85,105 @@ func (c *Controller) Process(ctx *ctrl.ProcessContext) (retriable bool, err erro
 		return false, nil
 	}
 
-	conflict, retriable, bundle, err := c.process(ctx.Logger, state)
+	bundle, external, conflict, retriable, err := c.process(ctx.Logger, state)
 	if conflict || bundle == nil && err == nil {
 		return false, nil
 	}
 
-	conflict, retriable, err = c.handleProcessResult(ctx.Logger, state, bundle, retriable, err)
+	conflict, processRetriable, processErr := c.handleProcessResult(ctx.Logger, state, bundle, retriable, err)
 	if conflict {
 		return false, nil
 	}
+	if processErr != nil {
+		if err != nil && !external {
+			ctx.Logger.Info("Failed to set State status", zap.Error(processErr))
+			return processRetriable || retriable, err
+		} else if external {
+			// We are going to return the error from handleProcessResult instead
+			ctx.Logger.Info("Invalid State Descriptor", zap.Error(err))
+		}
+		return processRetriable, processErr
+	}
 
-	return retriable, err
+	if err != nil {
+		if external {
+			ctx.Logger.Info("Invalid State Descriptor", zap.Error(err))
+			return false, nil
+		}
+		return retriable, err
+	}
+
+	return false, nil
 }
 
-// process processes the given State object performing autowiring for it.
-// It tries to return a Bundle even if there was an error.
-func (c *Controller) process(logger *zap.Logger, state *orch_v1.State) (conflictRet, retriableRet bool, b *smith_v1.Bundle, e error) {
+func (c *Controller) process(logger *zap.Logger, state *orch_v1.State) (b *smith_v1.Bundle, external bool, conflict bool, retriable bool, err error) {
+	entanglerContext, err := c.constructEntanglerContext(state)
+	if err != nil {
+		return nil, false, false, false, err
+	}
+
+	bundle, external, retriable, err := c.entangle(state, entanglerContext)
+	if err != nil {
+		return nil, external, false, retriable, err
+	}
+
+	conflict, retriable, bundle, err = c.createOrUpdateBundle(logger, state, bundle)
+	return bundle, false, conflict, retriable, err
+
+}
+
+func (c *Controller) constructEntanglerContext(state *orch_v1.State) (*wiring.EntangleContext, error) {
 	// Grab the namespace
 	namespaceObj, exists, err := c.NamespaceInformer.GetIndexer().GetByKey(state.Namespace)
 	if err != nil {
-		return false, false, nil, errors.WithStack(err)
+		return nil, errors.WithStack(err)
 	}
 	if !exists {
-		return false, false, nil, errors.Errorf("missing Namespace %q in informer", state.Namespace)
+		return nil, errors.Errorf("missing Namespace %q in informer", state.Namespace)
 	}
 	namespace := namespaceObj.(*core_v1.Namespace)
 
-	// Grab the ConfigMap
-	if state.Spec.ConfigMapName == "" {
```

Comment:
This should be in the schema

Reply:
Yep - I deleted this because it was already in the schema.

In fact, some of the wiring functions have checks which are already described in the schema too - I'll do those as part of another PR